pkg/dns/vips_allocator.go

```diff
@@ -128,6 +130,39 @@ func (d *VIPsAllocator) createOrUpdateMeshVIPConfig(
 	return d.persistence.Set(ctx, mesh, out)
 }
 
+func addFromMeshGateway(outboundSet *vips.VirtualOutboundMeshView, mesh string, gateway *core_mesh.MeshGatewayResource) {
+	for i, listener := range gateway.Spec.Conf.Listeners {
+		// We only setup outbounds for cross mesh listeners and only ones with a
+		// concrete hostname.
+		if !listener.CrossMesh || strings.Contains(listener.Hostname, "*") {
+			continue
+		}
+
+		for _, selector := range gateway.Spec.Selectors {
+			tags := mesh_proto.Merge(
+				gateway.Spec.GetTags(),
+				listener.GetTags(),
+				map[string]string{
+					"kuma.io/mesh": mesh,
+				},
+				selector.GetMatch(),
+			)
+			// TODO validate only one selector
```

Comment:
No, it's coming

Reply:
done

Reply:
@lahabana not a blocker for this PR but is this restriction actually necessary? Can multiple distinct outbound entries (different `TagSet` values) be added for the same hostname?

Without crossmesh it doesn't really come up because to expose your gateway, you must expose the `Dataplane`s and therefore always pick a set of tags somehow (with `MeshGatewayInstance` for example, you explicitly pick one set of tags and go through a `Service` + `Deployment`).

But when you stay inside the Kuma data plane, one could imagine exposing all the `Dataplane`s that match the selectors being the `hostname` and letting Envoy load balance over them, right?

```
selectors:
  - match:
      kuma.io/service: some-service
      version: v1
  - match:
      kuma.io/service: some-service
      version: v2
```